src/main.rs

```diff
@@ -9,78 +9,79 @@ fn main() -> io::Result<()> {
+                            print!("{}", todo)
                         }
+                        println!("\n")
                     }
                 }
                 Err(why) => eprintln!("Error reading file : {}", why),
-            }
-            println!("\n");
-
-            continue;
-        }
-
-        if input.trim() == "2" {
-            let mut new_todo = String::new();
-            println!("please input new todo >> \n");
-            io::stdout().flush()?;
-            io::stdin().read_line(&mut new_todo)?;
-
-            match try_add_lines(&new_todo) {
-                Err(why) => eprintln!("couldn't add {} : {}", new_todo, why),
-                Ok(_) => println!("success!\n"),
-            }
-            continue;
-        }
-
-        if input.trim() == "3" {
-            if try_read_lines("todo.txt")?.len() == 0 {
-                println!("Nothing to delete now \n");
+            },
+            "2" => {
+                let mut new_todo = String::new();
+                println!("please input new todo -> ");
                 io::stdout().flush()?;
-            } else {
-                println!("which one? >");
+                io::stdin().read_line(&mut new_todo)?;
 
-                if let Ok(vector) = try_read_lines("todo.txt") {
-                    for todo in vector {
-                        println!("{}", todo)
+                match try_add_lines(&new_todo) {
+                    Err(why) => eprintln!("couldn't add {} : {}", new_todo, why),
+                    Ok(_) => println!("success!\n"),
+                }
+            }
+            "3" => {
+                let to_dos = match try_read_lines("todo.txt") {
+                    Ok(lines) => lines,
+                    Err(e) => {
+                        eprintln!("Couldn't read file!! : {}", e);
+                        continue;
                     }
+                };
+
+                if to_dos.len() == 0 {
+                    println!("Nothing to delete now \n");
+                    io::stdout().flush()?;
                 } else {
-                    eprintln!("couldn't read file!!!")
-                }
+                    println!("You have below remained tasks");
+                    for todo in to_dos {
+                        println!("{}", todo);
+                    }
+                    print!("\n");
+                    print!("which one do you want to delete? ->");
+                    io::stdout().flush()?;
+                };
 
                 let mut delete_todo = String::new();
 
                 io::stdin().read_line(&mut delete_todo)?;
 
                 match find_index_opt(&delete_todo) {
-                    Some(_) => {
+                    Ok(_) => {
                         try_update_file(&delete_todo)?;
                         println!("successfully delete {}", delete_todo);
                     }
-                    None => println!("not found\n"),
+                    Err(why) => eprintln!("not found : {}", why),
```

Comment:
I feel this comment is not appropriate. In this case, the Err is returned when the file, todo.txt, is not found.

Reply:
Yes, I updated the text to tell it failed to read the file